tests/conftest.py

```diff
@@ -155,3 +155,87 @@ def upgrade_test_dir(test_dir):
     """
 
     return test_dir / "upgrade"
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    """
+    Hook to set the report of the test. Needed to check if a test failed.
+    """
+    outcome = yield
+    report = outcome.get_result()
+    if report.when == "call":
+        item.rep_call = report
+
+
+@pytest.fixture(scope="session")
+def shared_test_out_dir(tmp_path_factory):
+    """
+    Fixture to lazily create a shared temporary directory to store the dump
+    of MongoDB for failed tests and move them to store as an artifact is
+    running on github.
+    """
+    import os
+
+    _tmp_dir = None
+
+    def get_or_create():
+        nonlocal _tmp_dir
+        if _tmp_dir is None:
+            _tmp_dir = Path(tmp_path_factory.mktemp("shared_test_dir"))
+        return _tmp_dir
+
+    yield get_or_create
+
+    # If on github and the folder was created (i.e., some test failed), move
+    # it so that it can be uploaded as an artifact.
+    if os.getenv("GITHUB_WORKSPACE") and _tmp_dir:
+        artifact_path = Path(os.getenv("GITHUB_WORKSPACE")) / "test_folder"
+        os.rename(_tmp_dir, artifact_path)
+
+        print(f"Database dump saved as artifact: {artifact_path}")
```

Comment:
This will only be printed if pytest is run with -s no ? And as it is on github, we can't really change that "easily". Or maybe I missed some config somewhere that actually makes this print shown.

Reply:
Actually locally it is printed when a test fails and I think I checked that this is printed also on github.

Reply:
Ok